### PR TITLE
fix(cli): the fern cli appropriately generates examples for undiscriminated unions

### PR DESCRIPTION
**Pull request by dsinghvi**
We don't handle max depth properly in the ir examples generator.

### CODE REVIEW

File: packages/cli/generation/ir-generator/src/examples/generator/EndpointExampleFactory.ts
Comment (armandobelardo) [inline packages/cli/generation/ir-generator/src/examples/generator/EndpointExampleFactory.ts:1]: Nit: what's this?

Comment (dsinghvi) [reply packages/cli/generation/ir-generator/src/examples/generator/EndpointExampleFactory.ts:1]: adding this now (currently only type example generation)

File: packages/cli/generation/ir-generator/src/examples/ExampleGenerator.ts [hunk 1]
```diff
@@ -202,87 +209,138 @@ export class ExampleGenerator {
...
+
+        for (const pathParameter of rootPathParameters) {
+            const example = this.generatePathParameterExample({
+                pathParameter,
+                maybePathParameterExample: examples
+                    .flatMap((example) => example.rootPathParameters)
+                    .find((sp) => sp.name === pathParameter.name)
+            });
+            if (example == null) {
+                return undefined;
```
Comment (armandobelardo) [inline packages/cli/generation/ir-generator/src/examples/ExampleGenerator.ts:239]: Why return here? vs just adding if not null? Presumably you'd want to finish the loop and continue on?

File: packages/cli/generation/ir-generator/src/examples/ExampleGenerator.ts [hunk 2]
```diff
@@ -795,21 +913,28 @@ export class ExampleGenerator {
...
+        }
         return {
             jsonExample: exampleTypeReference.jsonExample,
-            shape: ExampleTypeReferenceShape.container(
-                ExampleContainer.optional({
-                    optional: exampleTypeReference,
-                    valueType: typeReference
-                })
-            )
+            shape: exampleTypeReference.shape
```
Comment (armandobelardo) [inline packages/cli/generation/ir-generator/src/examples/ExampleGenerator.ts:926]: Do you not need to wrap this like before?

File: packages/cli/generation/ir-generator/src/examples/ExampleGenerator.ts (hunk 1 above)
Comment (dsinghvi) [reply packages/cli/generation/ir-generator/src/examples/ExampleGenerator.ts:239]: that file is going to completely deleted now

File: packages/cli/generation/ir-generator/src/examples/ExampleGenerator.ts (hunk 2 above)
Comment (dsinghvi) [reply packages/cli/generation/ir-generator/src/examples/ExampleGenerator.ts:926]: this file is going to completely deleted now

**Pull request merged**